plugins/woocommerce/includes/tracks/events/class-wc-settings-tracking.php

```diff
@@ -143,13 +190,23 @@ public function track_setting_change( $option_name, $old_value, $new_value ) {
 	public function send_settings_change_event() {
 		global $current_tab, $current_section;
 
-		if ( empty( $this->updated_options ) ) {
+		if ( empty( $this->updated_options ) && empty( $this->deleted_options ) && empty( $this->added_options ) ) {
 			return;
 		}
 
-		$properties = array(
-			'settings' => implode( ',', $this->updated_options ),
-		);
+		$properties = array();
+
+		if ( ! empty( $this->updated_options ) ) {
+			$properties['settings'] = implode( ',', $this->updated_options );
+		}
+
+		if ( ! empty( $this->deleted_options ) ) {
+			$properties['deleted'] = implode( ',', $this->deleted_options );
+		}
+
+		if ( ! empty( $this->added_options ) ) {
+			$properties['added'] = implode( ',', $this->added_options );
```

Comment:
In my opinion I don't think we need to be this specific, and can skip the `added` and `deleted` options. Just adding `shipping_zone_method` or `shipping_class` to the `settings` array will be enough.
Instead you could do something similar as `modified_options` and have `modified_non_options` where we follow a format like this:
`'shipping_class' => 'add'`

cc: @pmcpinto this is for the `wcadmin_settings_change` event.

Reply:
> a format like this:
> 'shipping_class' => 'add'

I think this format is confusing, because it doesn't stay consistent. If for some reason a single request updates and adds/deletes at the same time, it would show like this:

```
'settings' => 'a, b, c',
'shipping_class' => 'add',
'other_setting' => 'delete
```

Can we maintain the current implementation for the settings change event? I think it's clearer.

I'm OK with using only one action for everything (`woocommerce_update_non_option_setting`)

Reply:
> I think this format is confusing, because it doesn't stay consistent. If for some reason a single request updates and adds/deletes at the same time, it would show like this

Hmm yeah that is a valid point. Your current approach does work better their.

> Can we maintain the current implementation for the settings change event? I think it's clearer.

Yeah that does make sense, lets keep your current format for the track event. One thing that I did notice though when testing was that shipping class add it shows this when adding a new shipping class: `settings: shipping_class_name,shipping_class_slug,shipping_class_description`, should we only cause this to happen when updating?
This is probably not as a big of a deal though.

> But I'm OK with using only one action for everything (`woocommerce_update_non_option_setting`)

Yeah I think that would still be great!